Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs

```diff
@@ -1489,5 +1490,551 @@ public void Including_an_interface_property_through_inheritance_should_work()
+            var expectation = new ClassWithNonBrowsableMembers() { AdvancedBrowsableProperty = 1 };
+
+            // Act
+            Action action =
+                () => subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void When_explicilty_browsable_field_differs_excluding_non_browsable_members_should_not_affect_result()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { ExplicitlyBrowsableField = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { ExplicitlyBrowsableField = 1 };
+
+            // Act
+            Action action =
+                () => subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void When_explicilty_browsable_property_differs_excluding_non_browsable_members_should_not_affect_result()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { ExplicitlyBrowsableProperty = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { ExplicitlyBrowsableProperty = 1 };
+
+            // Act
+            Action action =
+                () => subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void When_non_browsable_field_differs_excluding_non_browsable_members_should_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { NonBrowsableField = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { NonBrowsableField = 1 };
+
+            // Act & Assert
+            subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+        }
+
+        [Fact]
+        public void When_non_browsable_property_differs_excluding_non_browsable_members_should_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWithNonBrowsableMembers() { NonBrowsableProperty = 0 };
+            var expectation = new ClassWithNonBrowsableMembers() { NonBrowsableProperty = 1 };
+
+            // Act & Assert
+            subject.Should().BeEquivalentTo(expectation, config => config.ExcludingNonBrowsableMembers());
+        }
+
+        [Fact]
+        public void When_property_is_non_browsable_only_in_subject_excluding_non_browsable_members_should_not_make_it_succeed()
+        {
+            // Arrange
+            var subject = new ClassWhereMemberThatCouldBeNonBrowsableIsNonBrowsable() { PropertyThatMightBeNonBrowsable = 0 };
```

Comment:
🤔 Why is this property named with "might" even though it is marked with `EditorBrowsableState.Never`?

Reply:
The property has to have the exact same name in two different classes, and in one of them it is marked with `EditorBrowsableState.Never` and the other one it isn't.

Reply:
Ah yes, unless you use `WithMapping` of course ;-)

Reply:
Should I be doing that in these tests??

Reply:
Up to you. It may help understand the test better.

Reply:
I think it would make the tests more complex, and also mean that they're testing things beyond their scope.

Reply:
Debatable since it would have prevented me from asking the original question.

Reply:
But then the test is, in a very real sense, testing multiple things simultaneously. If `WithMapping` breaks, then these tests break too as a side-effect.

Reply:
That's a non-issue. We do that all the time when we use `Should().Throw()`